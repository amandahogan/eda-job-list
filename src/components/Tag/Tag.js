import React, { PropTypes } from 'react'
import { Label } from 'semantic-ui-react'

class Tag extends React.Component {
  render () {
    return (
      <Label
        as='a'
        color={this.props.category.colour}
        tag
      >{this.props.value}</Label>
    )
  }
}

Tag.propTypes = {
  category: PropTypes.shape({
    colour: PropTypes.string.isRequired
  }).isRequired,
  value: PropTypes.string.isRequired
}

export default Tag
