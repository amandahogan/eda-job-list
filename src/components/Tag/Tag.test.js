import { shallow } from 'enzyme'
import React from 'react'

import Tag from './Tag'

describe('<Tag />', () => {
  it('matches snapshot', () => {
    expect(
      shallow(<Tag category={{colour: 'red'}} value='test' />).getNode()
    ).toMatchSnapshot()
  })
})
