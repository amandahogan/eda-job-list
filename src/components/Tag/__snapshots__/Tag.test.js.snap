exports[`<Tag /> matches snapshot 1`] = `
<Label
  as="a"
  color="red"
  tag={true}>
  test
</Label>
`;
